MINOR: Fix tracing in KafkaApis.handle()

mimaison opened the pull request:
requestObj() returns null for the o.a.k.c.requests objects so use header() for these.

Once all the requests will have been replaced by o.a.k.c.requests objects, we should be able to clean that up, but in the meantime it's useful to trace both.

mimaison committed:
MINOR: Fix tracing in KafkaApis.handle()

requestObj returns null for the new Java request objects so use header for these.

ijuma commented on core/src/main/scala/kafka/server/KafkaApis.scala:
```diff
@@ -71,7 +71,7 @@ class KafkaApis(val requestChannel: RequestChannel,
     try {
       trace("Handling request:%s from connection %s;securityProtocol:%s,principal:%s".
-        format(request.requestObj, request.connectionId, request.securityProtocol, request.session.principal))
+        format(Option(request.requestObj).getOrElse(request.header), request.connectionId, request.securityProtocol, request.session.principal))
```
It may be worth doing it the other way around, as most requests are the new style now. Well spotted.

mimaison committed:
MINOR: Fix tracing in KafkaApis.handle()

Try to print header() first

mimaison commented:
Good point ! I've updated the PR

ijuma commented on core/src/main/scala/kafka/server/KafkaApis.scala:
```diff
@@ -71,7 +71,7 @@ class KafkaApis(val requestChannel: RequestChannel,
     try {
       trace("Handling request:%s from connection %s;securityProtocol:%s,principal:%s".
-        format(request.requestObj, request.connectionId, request.securityProtocol, request.session.principal))
+        format(Option(request.header).getOrElse(request.requestObj), request.connectionId, request.securityProtocol, request.session.principal))
```
Sorry, looking a bit deeper, I wonder if we should call the following method (in `RequestChannel.Request`):

``` scala
private def requestDesc(details: Boolean): String = {
      if (requestObj != null)
        requestObj.describe(details)
      else
        header.toString + " -- " + body.toString
    }
```

It's private, but maybe we should change that. What do you think?

mimaison committed:
MINOR: Fix tracing in KafkaApis.handle()

requestObj() returns null for the new Java request objects so instead use requestDesc()

mimaison commented:
I agree, that's probably the cleanest alternative. I've updated the PR accordingly.

ijuma commented on core/src/main/scala/kafka/server/KafkaApis.scala:
```diff
@@ -71,7 +71,7 @@ class KafkaApis(val requestChannel: RequestChannel,
     try {
       trace("Handling request:%s from connection %s;securityProtocol:%s,principal:%s".
-        format(request.requestObj, request.connectionId, request.securityProtocol, request.session.principal))
+        format(request.requestDesc(false), request.connectionId, request.securityProtocol, request.session.principal))
```
Thanks for updating. Is it correct to pass `false` here? I had a look at a few of the `toString` implementations and they all seemed to call `describe(true)`. Have you seen something different?

mimaison replied:
The argument of describe() is only used for old style queries and out of the last 2 still in use only Fetch makes use of it, so in the end I don't think it really matters. But for the sake of consistency, I'll change it to true

mimaison committed:
MINOR: Fix tracing in KafkaApis.handle()

requestObj() returns null for the new Java request objects so instead use requestDesc(true)

ijuma commented:
LGTM, thanks for your patience. :)

ijuma commented:
Merged to trunk and 0.10.0 branches.
